src/engine/sys/sdl_glimp.cpp

```diff
@@ -2033,9 +2033,27 @@ static void GLimp_InitExtensions()
 
 	logger.Notice("...using shading language version %i", glConfig2.shadingLanguageVersion );
 
-	// Texture formats and compression
+
+	// OpenGL driver constants.
+
+	glGetIntegerv( GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS, &glConfig2.maxTextureUnits );
+	glGetIntegerv( GL_MAX_TEXTURE_SIZE, &glConfig.maxTextureSize );
+	glGetIntegerv( GL_MAX_3D_TEXTURE_SIZE, &glConfig2.max3DTextureSize );
 	glGetIntegerv( GL_MAX_CUBE_MAP_TEXTURE_SIZE_ARB, &glConfig2.maxCubeMapTextureSize );
 
+	// Stubbed or broken drivers may have reported garbage.
+	glConfig.maxTextureSize = std::max( 0, glConfig.maxTextureSize );
```

Comment:
I have no idea, but that can't break anything I guess.

Reply:
If it's actually true that drivers report garbage, then we shouldn't disable features on the basis of said garbage.

Is it possible to check whether the 3D texture creation succeeded by any chance? If not maybe we'd be better off with detecting hardware/drivers.

Reply:
I removed that part. That very likely was a workaround for broken OpenGL 1 drivers in the nineties, drivers that we will never support for hardware that we will never support.

This workaround with this comment was already there in commit
https://github.com/id-Software/Quake-III-Arena/commit/dbe4ddb10315479fc00086f08e25d968b4b43c49 “_The Quake III Arena sources as originally released under the GPL license on August 20, 2005._”

If a driver in 2024 mistakenly reports 0 there (or a negative number we would clamp to 0), we would have a deepest problems than non-working lightgrid: no one texture would be renderable and I even don't know if the engine would survive this.

> Is it possible to check whether the 3D texture creation succeeded by any chance?

Yes that would be the same `GL_PROXY` mechanism that I use in this WIP PR (this one is for downscaling images if size is not accepted):

-  https://github.com/DaemonEngine/Daemon/pull/1131

Reply:
> > Is it possible to check whether the 3D texture creation succeeded by any chance?
> 
> Yes that would be the same `GL_PROXY` mechanism that I use in this WIP PR (this one is for downscaling images if size is not accepted):

I meant when you create it for real, not with the proxy mechanism.

Reply:
Hmm, it looks like I never submitted the comment I wrote. I was basically saying that I would prefer to not purposely use some feature that doesn't work and have the driver cry to the user about it.

For example Mesa currently prints things like that if we try to do 3D textures on VC4:

```
Mesa: User error: GL_INVALID_VALUE in glTexImage3D(level=0)
Mesa 22.3.6 implementation error: Bad t->Target in _mesa_test_texobj_completeness
Please report at https://gitlab.freedesktop.org/mesa/mesa/-/issues
```

Actually that third line telling the user to report a bug is wrong (the feature is not there, it will never be fixed, the bug is in the code attempting to use an unsupported feature), but the first line is totally legit, the user should not do something the driver/card doesn't support.

Also I experienced some crash with some driver versions before I've modified the engine to not attempt to do those things the hardware can't do. Of course it would be better if the driver would be more robust, but I cannot blame it for not behaving correctly if the mistake is purposely made by us.